### ISSUE TITLE
requirements / what should be implemented

### ISSUE THREAD

**Issue by bscheshirwork** [yiisoft/yii2-phpunit#1]
code
- tests/bootstrap.php - include require, exclude alias
- tests/compatibility.php - fix all of parent issues like this [14973](https://github.com/yiisoft/yii2/issues/14973), [126](https://github.com/yiisoft/yii2-httpclient/issues/126)
- tests/TestCase.php - general base class

docs
- about this repo / README.md
- how to set up travis configuration
- how to set up local tests 

> I have a guide-ru example for docs: [here](https://github.com/bscheshirwork/yii2-redis-session/blob/master/docs/tests.md)   
[codeception; see PhpStorm_Codeception_ReportPrinter_Redefine chapter](https://github.com/bscheshirwork/docker-yii2-app-advanced-rbac/blob/master/docs/codeception.md) for avoid output after header send

**Comment by bscheshirwork** [comment]
I think we need `docker-compose.yml` with `php included PHPUnit` example too

**Comment by samdark** [comment]
Not sure about docker (since it it can't be used without the phpunit itself and separately from tests) but agree about the rest.

**Comment by sergeymakinen** [comment]
https://github.com/sergeymakinen/tests could be useful although there are many "right" ways to accomplish this.

**Comment by rob006** [comment]
I don't think that we need `bootstrap.php` (it is project-specific) or `compatibility.php` (it is not required anymore). All we need is `TestCase` and custom assertions. Compatibility hacks should go directly to `TestCase` - this should be really easy if we bound extension version to phpunit version:
* 2.0.x requires phpunit 4, so `TestCase` contains only methods unavailable in phpunit 4 but available in phpunit 6.
* 2.1.x requires phpunit 5 and contains only methods unavailable in phpunit 5 but available in phpunit 6.
* 2.2.x requires phpunit 6 and contains only custom assertions.
